VirtoCommerce.Domain/Pricing/Model/PriceEvaluationContext.cs

```diff
@@ -14,7 +14,8 @@ public class PriceEvaluationContext : EvaluationContextBase
 		public string CatalogId { get; set; }
 		public string[] ProductIds { get; set; }
 		public string[] PricelistIds { get; set; }
-		public decimal Quantity { get; set; }
+	    public bool GetPricePerCurrencyByPriority { get; set; } = true;
```

Comment:
I would suggest defining the new flag for new behavior and call it `ReturnAllMatchedPrices = false`.
Instead of defining new flag for current behavior.

Reply:
Like so?

Reply:
Yes, thanks